src/state_migration/nv17/miner.rs

```diff
@@ -0,0 +1,925 @@
+                    let prev_in_sectors = fil_actors_shared::v8::Array::<
+                        fil_actor_miner_state::v8::SectorOnChainInfo,
+                        _,
+                    >::load(prev_in_root, store)?;
+                    let in_sectors = fil_actors_shared::v8::Array::<
+                        fil_actor_miner_state::v8::SectorOnChainInfo,
+                        _,
+                    >::load(in_root, store)?;
+                    let changes = fvm_ipld_amt::diff(&prev_in_sectors, &in_sectors)?;
+                    let mut prev_out_sectors = fil_actors_shared::v9::Array::<
+                        fil_actor_miner_state::v9::SectorOnChainInfo,
+                        _,
+                    >::load(prev_out_root, store)?;
+                    for change in changes {
+                        use fvm_ipld_amt::ChangeType;
+                        match &change.change_type {
+                            ChangeType::Remove => {
+                                prev_out_sectors.delete(change.key)?;
+                            }
+                            ChangeType::Modify | ChangeType::Add => {
+                                let info_v8 = in_sectors
+                                    .get(change.key)?
+                                    .context("Failed to get info from in_sectors")?;
+                                prev_out_sectors.set(
+                                    change.key,
+                                    TypeMigrator::migrate_type(info_v8.clone(), store)?,
+                                )?;
+                            }
+                        };
+                    }
+                    prev_out_sectors
+                } else {
+                    migrate_from_scratch(store, &in_array)?
+                }
+            } else {
+                migrate_from_scratch(store, &in_array)?
+            };
+
+            let out_root = out_array.flush()?;
+            cache.insert(miner_prev_sectors_in_key(miner_address), *in_root);
+            cache.insert(miner_prev_sectors_out_key(miner_address), out_root);
+
+            cache.insert(key, out_root);
+
+            Ok(out_root)
+        }
+    }
+
+    fn migrate_deadlines(
+        &self,
+        cache: &mut HashMap<String, Cid>,
+        store: &impl Blockstore,
+        deadlines: &Cid,
+    ) -> anyhow::Result<Cid> {
+        if deadlines == &self.empty_deadlines_v8_cid {
+            Ok(self.empty_deadlines_v9_cid)
+        } else {
+            let in_deadlines: fil_actor_miner_state::v8::Deadlines = store
+                .get_cbor(deadlines)?
+                .context("Failed to get in_deadlines")?;
+
+            let policy = match &self.network {
+                NetworkChain::Mainnet => fil_actors_shared::v9::runtime::Policy::mainnet(),
+                NetworkChain::Calibnet => fil_actors_shared::v9::runtime::Policy::calibnet(),
+                NetworkChain::Devnet(_) => unimplemented!("Policy::devnet"),
```

Comment:
It's not ideal. Theoretically we could contrive a definition somewhere in `fil-actor-states`, see https://github.com/ChainSafe/forest/blob/devnet-in-ci/src/networks/mod.rs#L210

That being said, it's not important at the moment. Maybe we could have a tracking issue instead.

Reply:
`Policy` under `v9` and `v10` are almost identical except one uses fvm2 types the other uses fvm3 types. We could maybe implement `From` trait between them to reuse `ChainConfig.policy`

Reply:
Tracking issue: https://github.com/ChainSafe/fil-actor-states/issues/151